zkevm-circuits/src/evm_circuit/param.rs

```diff
@@ -173,7 +174,14 @@ lazy_static::lazy_static! {
 }
 fn get_step_height_map() -> HashMap<ExecutionState, usize> {
     let mut meta = ConstraintSystem::<Fr>::default();
-    let circuit = EvmCircuit::configure(&mut meta);
+    let circuit = EvmCircuit::configure_with_params(
+        &mut meta,
+        FeatureConfig {
+            // Enable invalid_tx to get ExecutionState height
+            invalid_tx: true,
```

Comment:
What happens when we use this `step_height_map` with the feature `invalid_tx: false`?  Does any state height change?  Or is it only that we have an extra one for the InvalidTx state?
I think we could have the situation where a a state height changes, maybe not now?

I think it's worth adding a check to make sure that all common states between `step_height_map` built out of different `FeatureConfig`s have the same height, to detect bugs as soon as possible.

As an example. with the `invalid_tx` feature, the EndTx constraints change a bit.  Imagine this change causes an extra cell to be allocated, and then we have that `height[EndTx]=5` with `invalid_tx=false`, and `height[EndTx]=6` with `invalid_tx=true`!  Provably this is not the case right now, but I imagine this situation could happen with other features.